### DISCUSSION TITLE
Move heatmap generation in the UI

### DISCUSSION THREAD

**Discussion by sonakravtsenko3** [thygate/stable-diffusion-webui-depthmap-script#305] [General]
@thygate I think that "Generate HeatMap" should be replaced with an option "Use inferno colormap" under "Output DepthMap". This would de-clutter the UI. Downside is that getting heatmap and depthmap at the same time would require two generations. Alternatively, selecting this option could result in generating both depthmap and heatmap.

What do you think?

**Comment by thygate** [comment]
the current (and past) colormap is kind of a disappointment imho, It often does not use the entire range of the palette, and sticks to 1 or 2 colors, only barely improving visibility compared to a gray bitmap. I'm not even sure we should keep it tbh ... I'm not using it.

**Reply by sonakravtsenko3** [reply]
OK, will simply hide behind a setting in the WebUI extension settings. Thankfully should be trivial to do. I feel like deleting is only marginally easier.

**Reply by thygate** [reply]
Yeah, or as you initially suggested, under output depthmap, or remove all together .. all is fine with me 😄

**Comment by graemeniedermayer** [comment]
if the goal of heatmap was improving visibility. I really like rgb contour plots (my profile pic). Below you can clearly see 3 has the highest contrast whereas 1 and 2 have very flat faces.

![rgb_contours](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/38518120/2b525721-484c-4a27-aa51-0a367adbfe68)

using

```
import cv2
import numpy as np
import imageio

bins = 50
bins_range = [[256*256*i/bins, 256*256*(i+1)/bins]for i in range(bins)]
bin_h_colors = [i*180/bins for i in range(bins)]

img = np.array(cv2.imread("in.png", cv2.IMREAD_UNCHANGED))
h = np.array(img)
s = np.array(img)
l = np.array(img)

for i in range(bins):
    h[(bins_range[i][0] < h) & (bins_range[i][1] > h)] = bin_h_colors[i]
    s[:]= 255.9 
    l[(bins_range[i][0] < l) & (bins_range[i][1] > l)] = 256*(l[(bins_range[i][0] < l) & (bins_range[i][1] > l)] - bins_range[i][0])/(bins_range[i][1]-bins_range[i][0])

hls = np.dstack([h,l,s]).astype('uint8')

rgb_img = cv2.cvtColor(hls, cv2.COLOR_HLS2RGB)

imageio.imwrite("out.png",rgb_img)

```

**Reply by sonakravtsenko3** [reply]
The images on the left look more readable to my eyes. I guess this is a matter of habbit

**Reply by graemeniedermayer** [reply]
I should of included an starker example. To me this boost resnet looks okay but the contours make it really clear there's a lot of artefacts in this particular example.   
![resnet_contours](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/38518120/62c8f207-bf17-44bd-a95c-8c804188a920)

I don't really know if this is a necessary feature, but it can be helpful in understanding the differences between models.
